File: cf/191B.py
R=lambda:map(int,raw_input().split())
n,k=R()
m=R()[0]
c=R()
z=sorted(c[:-1])
s=sum(z[-k:])
print n if s <= m else 1+min(i for i,j in enumerate(c) if s-z[-k]+j>m)

File: cf/191B.pl
($n,$k)=split(/ /,<>);
$m=int <>;
@c=map{int $_} split(/ /,<>);
@z=sort {$a<=>$b}@c[0..$n-2];
$s += $z[$n-$_-1] for(1 .. $k);
$x=0;
print "$n\n" if $s <= $m;
for(0 .. $n-1){
	$x = $_ + 1 if $s - $z[$n-$k-1] + $c[$_] > $m;
	last if $x;
}
print "$x\n" if $s > $m;
